server/src/main/java/io/druid/segment/realtime/appenderator/BatchAppenderatorDriver.java

```diff
@@ -103,7 +103,7 @@ public AppenderatorDriverAddResult add(
       String sequenceName
   ) throws IOException
   {
-    return append(row, sequenceName, null, false, true);
+    return append(row, sequenceName, null, true, true);
```

Comment:
Why is the BatchAppenderatorDriver skipping the lineage check now? I thought it could still make more than one segment per interval if it's running in non-incremental-publishing mode.